### PR TITLE
feat: update events to accomodate accounts

### LINKED ISSUE #7840: [Task]: Update events to accomodate accounts
### Task description

The wallet receives events from the SDK notifying about changes, we should update this code to accomodate the account centric model, if necessary

### LINKED ISSUE #7862: [Task]: Update wallet events to accomodate updates
### Task description

Update SDK events received by firefly

### PR DESCRIPTION
**Pull request by cpl121**
Closes #7840
Closes #7862

## Summary

> Please summarize your changes, describing **what** they are and **why** they were made.

...

## Changelog

```
- Please write a list of changes
```

### Platforms

> Please select any platforms where your changes have been tested.

-   **Desktop**
    -   [ ] MacOS
    -   [ ] Linux
    -   [ ] Windows

### Instructions

> Please describe the specific instructions, configurations, and/or test cases necessary to **test and verify** that your changes work as intended.

...

## Checklist

> Please tick the following boxes that are relevant to your changes.

-   [ ] I have followed the contribution guidelines for this project
-   [ ] I have performed a self-review of my own code
-   [ ] I have commented my code, particularly in hard-to-understand areas
-   [ ] I have added or modified tests that prove my changes work as intended
-   [ ] I have verified that new and existing unit tests pass locally with my changes
-   [ ] I have verified that my latest changes pass CI workflows for testing and linting
-   [ ] I have made corresponding changes to the documentation

### CODE REVIEW

File: packages/shared/lib/core/wallet/utils/validateWalletApiEvent.ts
Comment (marc2332) [inline packages/shared/lib/core/wallet/utils/validateWalletApiEvent.ts:1]: I casually ended up fixing some of this code in my profile manager refactor PR

key points: 
- Events do no longer include any sort of ID, this must be passed manually, example: https://github.com/iotaledger/firefly/pull/7809/files#diff-b497998dedc3783a901118c29dd2b0064ac8d300f824684007ee90d10beaab38
- `Event` type is gone, use `WalletEvent`.
- Example with [`handleNewOutputEvent`](https://github.com/iotaledger/firefly/pull/7809/files#diff-5790b75f3a351fa8556f6c427a234a781582ce1342b29dd6914d80c2da3fba84)
- [`validateWalletApiEvent`](https://github.com/iotaledger/firefly/pull/7809/files#diff-bdd4828996ff171206b613773bb132f2f640e745c415c629f84669c5f1714113)

**Review by marc2332 — CHANGES_REQUESTED** [review]

File: packages/desktop/views/dashboard/wallet/views/implicit-account-creation-multistep/AccountCreationView.svelte [hunk 1]
```diff
@@ -15,6 +17,8 @@
             message: localize('error.global.generic'),
         })
     }
+
+    onDestroy(unsubscribeFromWalletApiEvents)
```
Comment (marc2332) [inline packages/desktop/views/dashboard/wallet/views/implicit-account-creation-multistep/AccountCreationView.svelte:21]: let's put this on `ImplicitAccountCreationView.svelte`.

**Review by marc2332 — CHANGES_REQUESTED** [review]

File: packages/shared/lib/contexts/wallet/implicit-account-creation/actions/handleNewOutputForImplicitAccountCreation.ts [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+export function handleNewOutputForImplicitAccountCreation(walletId: string): WalletApiEventHandler {
+    return (error: Error, rawEvent: WalletEvent) => {
+        validateWalletApiEvent(error, rawEvent, WalletEventType.NewOutput)
+        if (rawEvent.type === WalletEventType.TransactionInclusion) {
+            handleNewOutputForImplicitAccountCreationInternal(walletId, rawEvent as NewOutputWalletEvent)
+        }
+    }
+}
+
+export async function handleNewOutputForImplicitAccountCreationInternal(
```
Comment (marc2332) [inline packages/shared/lib/contexts/wallet/implicit-account-creation/actions/handleNewOutputForImplicitAccountCreation.ts:15]: Let's get rid of this, I would move this logic to `handleNewOutputEventInternal.ts`

File: packages/desktop/views/dashboard/wallet/views/implicit-account-creation-multistep/AccountCreationView.svelte (hunk 1 above)
Comment (marc2332) [reply packages/desktop/views/dashboard/wallet/views/implicit-account-creation-multistep/AccountCreationView.svelte:21]: also remove this line

File: packages/desktop/views/dashboard/wallet/views/ImplicitAccountCreationView.svelte [hunk 3]
```diff
@@ -1,10 +1,14 @@
...
+    import { onMount } from 'svelte'
     import { implicitAccountCreationRoute, ImplicitAccountCreationRoute } from '../index'
     import { InitView, AccountCreationView, FundConfirmationView, OneTimeDepositView } from '.'
     import { Text, TextType } from 'shared/components'
     import { localize } from '@core/i18n'
+    import { subscribeToWalletApiEventsForImplicitAccountCreation } from '@contexts/wallet/implicit-account-creation'
 
     const IMPLICIT_ACCOUNT_STEPS = Object.keys(ImplicitAccountCreationRoute).slice(1)
+
+    onMount(subscribeToWalletApiEventsForImplicitAccountCreation)
```
Comment (marc2332) [inline packages/desktop/views/dashboard/wallet/views/ImplicitAccountCreationView.svelte:11]: and this one

File: packages/shared/lib/contexts/wallet/implicit-account-creation/actions/subscribeToWalletApiEventsForImplicitAccountCreation.ts
Comment (marc2332) [inline packages/shared/lib/contexts/wallet/implicit-account-creation/actions/subscribeToWalletApiEventsForImplicitAccountCreation.ts:1]: and this as well

**Review by marc2332 — CHANGES_REQUESTED** [review]

**Review by marc2332 — APPROVED** [review]
:fire:

File: packages/desktop/views/dashboard/wallet/views/ImplicitAccountCreationView.svelte (hunk 3 above)
[the remark this answers is not in the document]
Comment (marc2332) [reply packages/desktop/views/dashboard/wallet/views/ImplicitAccountCreationView.svelte]: that's old code, refresh the changes

File: packages/shared/lib/core/wallet/actions/events-handlers/handleNewOutputEvent.ts [hunk 4]
```diff
@@ -42,6 +52,12 @@ export async function handleNewOutputEventInternal(walletId: string, payload: Ne
         !get(allWalletActivities)[walletId].find((_activity) => _activity.id === outputData.outputId)
     const isNftOutput = output.type === OutputType.Nft
 
+    const isImplicitAccountOutput =
```
Comment (begonaalvarezd) [inline packages/shared/lib/core/wallet/actions/events-handlers/handleNewOutputEvent.ts:55]: its missing to check that the address unlock condition belongs to the implicit account address, you can do so  by checking that the address unlock condition is your implicit account creation address, or by checking the type of that address

**Review by begonaalvarezd — CHANGES_REQUESTED** [review]

**Review by begonaalvarezd — APPROVED** [review]

**Pull request merged**